src/pages/Opportunities.jsx

```diff
@@ -63,29 +104,38 @@ function Opportunities() {
                 key={index}
               >
                 <div className="mb-4 md:mb-0 w-full md:w-48 md:h-32 flex-shrink-0 md:mx-0 md:mr-4">
-                  <a href={`/opportunities/${el.id}`}>
+                  <Link
+                    to={`/opportunities/${el.id}`}
+                    state={{ opportunityData: el }}
```

Comment:
I would avoid relying on passing the state to the page like this. More details in my comment on the `OpportunityDetail` component below